spec/Section 6 -- Execution.md

```diff
@@ -143,6 +147,131 @@ ExecuteMutation(mutation, schema, variableValues, initialValue):
+observed. As an example, a "Pub-Sub" system may produce an event stream when
+"subscribing to a topic", with an event occurring on that event stream for each
+"publish" to that topic. Event streams may produce an infinite sequence of
+events or may complete at any point. Event streams may complete in response to
+an error or simply because no more events will occur. An observer may at any
+point decide to stop observing an event stream, after which it must receive no
+more events from that event stream.
+
+Note: If an event stream's observer has stopped observing, that may be a good
+opportunity to clean up any associated resources such as closing any connections
+which are no longer necessary.
+
+#### Subscribe
+
+Executing a subscription creates a persistent function on the server that
+maps an underlying event stream to the Publish Stream. The logic to create the
+underlying event stream is application-specific and takes the root field and
+query variables as inputs.
+
+Subscribe(schema, subscription, operationName, variableValues, initialValue):
+
+  * Let {subscriptionType} be the root Subscription type in {schema}.
+  * Assert: {subscriptionType} is an Object type.
+  * Let {selectionSet} be the top level Selection Set in {subscription}.
+  * Let {sourceStream} be the result of running {CreateSourceEventStream(schema, subscription, selectionSet, variableValues, initialValue)}.
+  * Let {responseStream} be the result of running
+    {MapSourceToResponseEvent(sourceStream)}
+  * Return {responseStream}.
+
+CreateSourceEventStream(schema, subscriptionType, selectionSet, variableValues,
+  initialValue):
+
+  * Let {rootField} be the first top level field in {selectionSet}.
+  * Let {argumentValues} be the result of {CoerceArgumentValues(subscriptionType, rootField, variableValues)}.
+  * Let {fieldStream} be the result of running {ResolveFieldEventStream(subscriptionType, initialValue, rootField, argumentValues)}.
+  * Return {fieldStream}.
+
+ResolveFieldEventStream(subscriptionType, rootValue, fieldName, argumentValues):
+  * Let {resolver} be the internal function provided by {subscriptionType} for
+    determining the resolved value of a field named {fieldName}.
+  * Return the result of calling {resolver}, providing {rootValue} and {argumentValues}.
+
+#### Publish Stream
+
+Each event in the underlying event stream triggers execution of the subscription
+selection set.
+
+MapSourceToResponseEvent(sourceStream, subscriptionType, selectionSet, variableValues):
+
+  * Return a new event stream {publishStream} which yields events as follows:
+  * For each {event} on {sourceStream}:
+    * Let {publishPayload} be the result of running
+      {ExecuteSelectionSet(selectionSet, subscriptionType, event, variableValues)}
+      *normally* (allowing parallelization).
+    * Let {errors} be any *field errors* produced while executing the
+      selection set.
+    * Let {response} be an unordered map containing {publishPayload} and, optionally,
```

Comment:
this is a bit ambiguous since we expect the unordered map to use the key `data` but this implies using a key `publishPayload` - maybe just rename the variable